internal/bminventory/inventory.go

```diff
@@ -497,6 +499,34 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 	return b.GetClusterInternal(ctx, installer.GetClusterParams{ClusterID: *cluster.ID})
 }
 
+func (b *bareMetalInventory) isOpenshiftVersionSupported(openshiftversion string) bool {
+	ocpVersionKey, err := b.versionsHandler.GetKey(openshiftversion)
+	if err != nil {
+		return false
+	}
+
+	return b.versionsHandler.IsOpenshiftVersionSupported(ocpVersionKey)
+}
+
+// Returns version as fetched from release image metadata (used for storing in DB as 'OpenshiftVersion')
+func (b *bareMetalInventory) getReleaseVersion(openshiftversion string) string {
+	version, err := b.versionsHandler.GetReleaseVersion(openshiftversion)
+	if err != nil {
+		return openshiftversion
+	}
+	return version
+}
+
+// Returns release image from version (used for storing in DB as 'OcpReleaseImage')
+func (b *bareMetalInventory) getReleaseImage(openshiftversion string) string {
+	releaseImage, err := b.versionsHandler.GetReleaseImage(openshiftversion)
+	if err != nil {
+		// Shouldn't happen, supported version is verified in RegisterCluster
+		return ""
+	}
+	return releaseImage
+}
```

Comment:
I think that this is too much logic that should be handled on `versions.go`. No?

Reply:
Yeah thought about it, but the are just helpers to avoid cyclomatic complexity error in RegisterClusterInternal.
Probably don't want to change the API versions just for that. Can move them to versions as private funcs, but then it'll be used only here, not that nice:/ WDYT? worth adding some new APIs for it maybe?